spec/02-integration/02-cmd/02-start_stop_spec.lua

```diff
@@ -39,6 +41,102 @@ describe("kong start/stop", function()
+    --[[
+    What are we testing exactly?
+    When Kong starts, the cli commands will shell out to start the 'nginx'
+    executable. The stdout/stderr of this command is redirected to be captured
+    in case of errors.
+    A problem arises when Kong runs in the foreground 'nginx_daemon=off', and
+    the nginx log output is set to '/dev/stderr' and '/std/stdout'.
+    In this case the shell command to start Kong will capture all output send
+    to stdout/stderr by nginx. Which causes the tempfiles to grow uncontrolable.
+
+    So when Kong is run in the foreground, no redirects should be used when
+    starting Kong to prevent the above from happening. Or in other words we
+    want the stdout/stderr of the "kong start" command to receive all output
+    of the nginx logs. Instead of them being swallowed by the intermediary
+    nginx start command (used by "kong start" under the hood).
+    --]]
+    local xos = _G.os  -- trick the linter here in accepting global updates
+    local exec = xos.execute
+    local stdout = pl_path.tmpname()
+    local stderr = pl_path.tmpname()
+    -- create a finalizer that will stop Kong and cleanup logfiles
+    finally(function()
+      os.remove(stdout)
+      os.remove(stderr)
+      xos.execute = exec
+      assert(helpers.kong_exec("stop", {
+        prefix = helpers.test_conf.prefix,
+      }))
+    end)
+    -- catch the prepared start command (let the 'helpers' do
+    -- the hard work of building the command)
+    local start_cmd
+    xos.execute = function(cmd) start_cmd = cmd return true end
+    assert(helpers.kong_exec("start", {
+      prefix = helpers.test_conf.prefix,
+      database = helpers.test_conf.database,
+      pg_database = helpers.test_conf.pg_database,
+      cassandra_keyspace = helpers.test_conf.cassandra_keyspace,
+      proxy_listen = "127.0.0.1:" .. helpers.test_conf.proxy_port,
+      admin_listen = "127.0.0.1:" .. helpers.test_conf.admin_port,
+      admin_access_log = "/dev/stdout",
+      admin_error_log = "/dev/stderr",
+      proxy_access_log = "/dev/stdout",
+      proxy_error_log = "/dev/stderr",
+      log_level = "debug",
+      nginx_daemon = "off",
+    }))
+    xos.execute = exec
+    -- remove the stdout/stderr redirects from the captured command
+    -- and insert new ones for us to track
+    start_cmd = start_cmd:match("^(.- bin/kong start).-$")
+    start_cmd = start_cmd .. " > " .. stdout .. " 2> " .. stderr
+    start_cmd = start_cmd .. " &"  -- run it detached
+    -- Now start Kong non-daemonized, but detached
+    assert(pl_utils.execute(start_cmd))
+    -- wait for Kong to be up and running, create a test api
+    helpers.wait_until(function()
+      local success, admin_client = pcall(helpers.admin_client)
+      if not success then
+        --print(admin_client)
+        return false
+      end
+      local res = assert(admin_client:send {
+        method = "POST",
+        path = "/apis",
```

Comment:
0.13 is released! :) we shouldn't use the deprecated `/apis` entity in new tests

Reply:
Yes noticed that too.